openmc/mgxs_library.py

```diff
@@ -582,14 +626,43 @@ def nu_fission(self, nu_fission):
                         [self.vector_shape, self.matrix_shape])
             # Find out if we have a nu-fission matrix or vector
             # and set a flag to allow other methods to check this later.
-            self.use_chi = (npnu_fission.shape == self.vector_shape)
+            if npnu_fission.shape == self.vector_shape:
+                self.use_chi = True
+            else:
+                self.use_chi = False
 
-        self._nu_fission = npnu_fission
-        if np.sum(self._nu_fission) > 0.0:
+        i = self.temperatures.tolist().index(temperature)
+        self._nu_fission[i] = npnu_fission
+        if np.sum(npnu_fission) > 0.0:
```

Comment:
ditto, since the `fissionable` attribute is set here, why give the user an option to set the attribute themselves?